### ISSUE TITLE
Smarter handling of columns tied to indexes, constraints, and foreign keys

### ISSUE THREAD

**Issue by rscs** [dbeaver/dbeaver#2958]
I tried to delete two columns from a table ...

```
ALTER TABLE eliasqadb.matrix_rates DROP COLUMN supplierId ;
ALTER TABLE eliasqadb.matrix_rates DROP COLUMN fileDate ;
```

... and I got the error:

```
SQL Error [1553] [HY000]: Cannot drop index 'matrix_rates_matrix_suppliers_FK': needed in a foreign key constraint
  Cannot drop index 'matrix_rates_matrix_suppliers_FK': needed in a foreign key constraint
  Cannot drop index 'matrix_rates_matrix_suppliers_FK': needed in a foreign key constraint
```

Okay, so then I head over to the "Foreign Keys" tab and delete the associated entry, which generates the following SQL:

```
ALTER TABLE eliasqadb.matrix_rates DROP COLUMN supplierId ;
ALTER TABLE eliasqadb.matrix_rates DROP COLUMN fileDate ;
ALTER TABLE eliasqadb.matrix_rates DROP FOREIGN KEY matrix_rates_matrix_suppliers_FK ;
```

But this still fails because the ```DROP FOREIGN KEY``` needs to come BEFORE the ```DROP COLUMN``` in order for this to work.  So I revert all of my changes and delete the foreign key row from the GUI first, then delete the two columns via the GUI and finally the SQL is correct:

```
ALTER TABLE eliasqadb.matrix_rates DROP FOREIGN KEY matrix_rates_matrix_suppliers_FK ;
ALTER TABLE eliasqadb.matrix_rates DROP COLUMN supplierId ;
ALTER TABLE eliasqadb.matrix_rates DROP COLUMN fileDate ;
```

It would be great if DBeaver was smarter about this and did two things:

1 - Automatically dropped indexes, constraints, and foreign keys when their associated columns are dropped. Or at least give some kind of warning when hitting "Save" before the SQL is generated.

2 - Automatically ordered these DDL operations for me instead of me having to know what order to do them in.

**Comment by serge-rider** [comment]
Agreed. Although it will require to improve whole metadata editor back-end model. 
Currently it is quite simple.

**Comment by uslss** [comment]
verified

**Comment by LonwoLonwo** [comment]
Implemented in 21.0.1

Now in case of column delete DBeaver will search all dependent objects for this column (PK, UK, FK and indexes) and suggest remove these dependent objects to. In case, if you will choose "No" in this "delete dependent objects" dialog, Dbeaver will not delete dependent objects nor column.

![111466294-2abcd680-8734-11eb-8e25-e489f01c47f3](https://user-images.githubusercontent.com/45152336/111595808-8f803b80-87dd-11eb-9a95-341b4b61888c.png)
